### ISSUE TITLE
Rendering website with `reveal.js` slides

### ISSUE THREAD

**Issue by JulianTao** [quarto-dev/quarto-cli#216]
I created a website for a course. The output formats were defined in the `_quarto.yml` for the project as follows:

```{yaml}
format:
  html:
    theme: 
      - cosmo
      - ./assets/mystyles.scss
    toc: true
  revealjs:
    theme: ./assets/myrevealjs.scss
    slideNumber: true
    history: false
    width: 1600
    height: 900
    slide-level: 2
    html-math-method: katex
    chalkboard:
      theme: whiteboard
```
Website pages include a series of  `lecture.qmd` and  an `index.qmd` with links to the lectures. The idea is that students and instructors can go to the website and click any lecture on the front page to view the corresponding `revealjs` slides.
 
For each lecture `qmd` file, I included `format: revealjs` in the YAML header; in the `index.qmd` file, I included `format: html`.

I expect that when I run `quarto render` in the project directory, the `index.qmd` should be rendered to `html` and all the lectures will be rendered into `revealjs`. However, all the files are rendered into conventional `html` with the `cosmo` theme.

For now, I created a bash script to render the site using a single command. In the script, all the files to first rendered to`revealjs`, the `index.qmd` file is then rendered into the conventional `html`.

It will be helpful if I just need to run `quarto render` to render the site properly. I believe it is supposed to work like that.

**Comment by jjallaire** [comment]
We did do some work recently to make exactly what you describe above possible, so there is definitely a bug here. I can repro this not working as expected locally, but it behaves a bit differently than what you describe above (I actually get an error when rendering the site). What version of Quarto are you running?

**Comment by jjallaire** [comment]
This commit should resolve the issue and get the `quarto render` command working as you expect: https://github.com/quarto-dev/quarto-cli/commit/85d2dd051867df94ca693ccc3b1c8be146981a40. This fix is available in v0.2.349

**Comment by JulianTao** [comment]
> We did do some work recently to make exactly what you describe above possible, so there is definitely a bug here. I can repro this not working as expected locally, but it behaves a bit differently than what you describe above (I actually get an error when rendering the site). What version of Quarto are you running?

As you said, there is an error warning when render the project directly. I now remember that that's why I created a script.

It is fixed with the latest update. Thank you!
